graph/src/abi/event_ext.rs

```diff
@@ -0,0 +1,53 @@
+use alloy::json_abi::Event;
+use alloy::primitives::LogData;
+use anyhow::anyhow;
+use anyhow::Context;
+use anyhow::Result;
+use itertools::Itertools;
+use web3::types::Log;
+
+use crate::abi::DynSolParam;
+
+pub trait EventExt {
+    fn decode_log(&self, log: &Log) -> Result<Vec<DynSolParam>>;
```

Comment:
would be great to make sure this is tested, it's ok if there's already tests elswhere, I'm assuming this is for backwards compatibility. Although I would probably still like to have some local tests for these impls